[BUG] org.opensearch.remotestore.RemoteStoreStatsIT.testStatsOnShardUnassigned is flaky

**Describe the bug**
org.opensearch.remotestore.RemoteStoreStatsIT.testStatsOnShardUnassigned test is flaky on main branch. Encountered this failure during PR build of #8758. 

**To Reproduce**
REPRODUCE WITH: 
```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.RemoteStoreStatsIT.testStatsOnShardUnassigned" -Dtests.seed=1EE76B3CC8051A6 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=hu-HU -Dtests.timezone=Africa/Bissau -Druntime.java=20
```

**Expected behavior**
Test should pass.

**Plugins**
Please list all plugins currently enabled.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - OS: [e.g. iOS]
 - Version [e.g. 22]

**Additional context**
Jenkins - https://build.ci.opensearch.org/job/gradle-check/21584/testReport/junit/org.opensearch.remotestore/RemoteStoreStatsIT/testStatsOnShardUnassigned/

Seeing failures due to cluster state update task leading to exception - 
```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.remotestore.RemoteStoreStatsIT.testStatsOnShardUnassigned" -Dtests.seed=5330C7A06AF381B5 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ms -Dtests.timezone=Africa/Nouakchott -Druntime.java=20

org.opensearch.remotestore.RemoteStoreStatsIT > testStatsOnShardUnassigned FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=70, name=opensearch[node_t1][clusterApplierService#updateTask][T#1], state=RUNNABLE, group=TGRP-RemoteStoreStatsIT]
        at __randomizedtesting.SeedInfo.seed([5330C7A06AF381B5:A5FA04C1C02A5874]:0)

        Caused by:
        java.lang.AssertionError: a started primary with non-pending operation term must be in primary mode [remote-store-test-idx-1][0], node[3dgi0k4bQvm6tE_udLNSlA], [P], s[STARTED], a[id=oOnPRo_VQYu46NzG2Kr5GA]
            at __randomizedtesting.SeedInfo.seed([5330C7A06AF381B5]:0)
            at org.opensearch.index.shard.IndexShard.updateShardState(IndexShard.java:735)
            at org.opensearch.indices.cluster.IndicesClusterStateService.updateShard(IndicesClusterStateService.java:714)
            at org.opensearch.indices.cluster.IndicesClusterStateService.createOrUpdateShards(IndicesClusterStateService.java:654)
            at org.opensearch.indices.cluster.IndicesClusterStateService.applyClusterState(IndicesClusterStateService.java:297)
            at org.opensearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:606)
            at org.opensearch.cluster.service.ClusterApplierService.callClusterStateAppliers(ClusterApplierService.java:593)
            at org.opensearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:561)
            at org.opensearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:484)
            at org.opensearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:186)
            at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:849)
            at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:282)
            at org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:245)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
            at java.base/java.lang.Thread.run(Thread.java:1623)
```

--- a/server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreStatsIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreStatsIT.java
@@ -9,6 +9,7 @@
 package org.opensearch.remotestore;
 
 import org.junit.Before;
+import org.opensearch.action.admin.cluster.health.ClusterHealthResponse;
 import org.opensearch.action.admin.cluster.remotestore.restore.RestoreRemoteStoreRequest;
 import org.opensearch.action.admin.cluster.remotestore.stats.RemoteStoreStats;
 import org.opensearch.action.admin.cluster.remotestore.stats.RemoteStoreStatsRequestBuilder;
@@ -22,6 +23,7 @@ import org.opensearch.cluster.routing.allocation.command.MoveAllocationCommand;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.index.IndexSettings;
 import org.opensearch.index.remote.RemoteSegmentTransferTracker;
+import org.opensearch.test.InternalTestCluster;
 import org.opensearch.test.OpenSearchIntegTestCase;
 
 import java.io.IOException;
@@ -443,12 +445,19 @@ public class RemoteStoreStatsIT extends RemoteStoreBaseIntegTestCase {
         createIndex(INDEX_NAME, remoteStoreIndexSettings(2, 1));
         ensureGreen(INDEX_NAME);
         indexDocs();
-        int dataNodeCountBeforeStop = client().admin().cluster().prepareHealth().get().getNumberOfDataNodes();
-        internalCluster().stopRandomDataNode();
+        ClusterHealthResponse clusterHealthResponse = client().admin().cluster().prepareHealth().get();
+        int dataNodeCountBeforeStop = clusterHealthResponse.getNumberOfDataNodes();
+        int nodeCount = clusterHealthResponse.getNumberOfNodes();
+        String nodeToBeStopped = randomBoolean() ? primaryNodeName(INDEX_NAME) : replicaNodeName(INDEX_NAME);
+        internalCluster().stopRandomNode(InternalTestCluster.nameFilter(nodeToBeStopped));
+        ensureYellowAndNoInitializingShards(INDEX_NAME);
+        ensureStableCluster(nodeCount - 1);
         RemoteStoreStatsResponse response = client().admin().cluster().prepareRemoteStoreStats(INDEX_NAME, "0").get();
         int dataNodeCountAfterStop = client().admin().cluster().prepareHealth().get().getNumberOfDataNodes();
         assertEquals(dataNodeCountBeforeStop, response.getTotalShards());
         assertEquals(dataNodeCountAfterStop, response.getSuccessfulShards());
+        // Indexing docs to ensure that the primary has started
+        indexSingleDoc(INDEX_NAME);
     }
 
     public void testStatsOnRemoteStoreRestore() throws IOException {

